plugins/org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/incremental/BuildRequest.xtend

```diff
@@ -75,17 +42,47 @@ class BuildRequest {
 	IPostValidationCallback afterValidate = new DefaultValidationCallback()
 	(URI, URI)=>void afterGenerateFile = []
 	(URI)=>void afterDeleteFile = []
-	(URI)=>boolean belongsToThisBuildRun = [true]
 	
 	def boolean belongsToThisBuildRun(URI uri) {
-		return belongsToThisBuildRun.apply(uri)
+		return uri.toString.startsWith(baseDir.toString)
```

Comment:
was intended to be an overridable default, but I now removed the whole thing, since it is now expected that only local ResourceDescription are passed through the ResourceDescritpionsData